hal/halx86/include/halp.h

```diff
@@ -185,6 +185,14 @@ typedef struct _HalAddressUsage
     } Element[];
 } ADDRESS_USAGE, *PADDRESS_USAGE;
 
+typedef struct _AMLI_ILLEGAL_IO_PORT_ADDRESSES
```

Comment:
Please use the public structure `HAL_AMLI_BAD_IO_ADDRESS_LIST` from `ntddk.h` instead.

Reply:
Fixed, thanks.
Also, since there was incompatible pointer types between the 4th parameter of `PHALIOREADWRITEHANDLER`, which is used in `HAL_AMLI_BAD_IO_ADDRESS_LIST` and the 4th parameter of `HaliHandlePCIConfigSpaceAccess`, I got the compilation error. So I changed the parameter of the last one from `PVOID` to `PULONG`. I guess it will be more correct, in case if that array should use the public structure instead of private one declared by Vadim.